src/Security.php

```diff
@@ -55,4 +55,21 @@ public static function wp_redirect($url) {
     return $url;
   }
 
+  /**
+   * @implemements wp_headers
+   *
+   * @return array
+   */
+  public static function additional_securityheaders(array $headers):array {
+    if (!is_admin()) {
+      $headers['Referrer-Policy']           ??= defined('CORE_STANDARDS_Referrer_Policy') ? CORE_STANDARDS_Referrer_Policy : 'no-referrer-when-downgrade';
+      $headers['X-Content-Type-Options']    ??= defined('CORE_STANDARDS_X_Content_Type_Options') ? CORE_STANDARDS_X_Content_Type_Options : 'nosniff';
+      $headers['X-XSS-Protection']          ??= defined('CORE_STANDARDS_X_XSS_Protection') ? CORE_STANDARDS_X_XSS_Protection : '1; mode=block';
+      $headers['Permissions-Policy']        ??= defined('CORE_STANDARDS_Permissions_Policy') ? CORE_STANDARDS_Permissions_Policy : 'geolocation=(self "https://example.com") microphone=() camera=()';
+      $headers['Content-Security-Policy']   ??= defined('CORE_STANDARDS_Content_Security_Policy') ? CORE_STANDARDS_Content_Security_Policy : 'frame-ancestors';
```

Comment:
It is my understanding that the Content Security Policy header replaces most of the other headers, such as X-Frame-Options. Shouldn't we skip the legacy headers and directly implement all directives in the CSP instead?

Reply:
I can not find any documentation about what you state here, but I trust you if you want me to do it like that, could you please just tell which other headers are replaced by this one?

Reply:
I googled "migrate x-frame-options to csp frame-ancestors" and found this:
https://stackoverflow.com/questions/43039706/replacing-x-frame-options-with-csp

And https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection states that it is deprecated and not supported by some browsers:
> it is recommended that you use [Content-Security-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy) without allowing unsafe-inline scripts instead.